gateway/src/main/java/org/georchestra/gateway/filter/headers/providers/JsonPayloadHeadersContributor.java

```diff
@@ -64,21 +65,20 @@ public JsonPayloadHeadersContributor() {
     }
 
     public @Override Consumer<HttpHeaders> prepare(ServerWebExchange exchange) {
-        return headers -> {
-            GeorchestraTargetConfig.getTarget(exchange)//
-                    .map(GeorchestraTargetConfig::headers)//
-                    .ifPresent(mappings -> {
-                        Optional<GeorchestraUser> user = GeorchestraUsers.resolve(exchange);
-                        Optional<Organization> org = GeorchestraOrganizations.resolve(exchange);
+        return headers -> GeorchestraTargetConfig.getTarget(exchange)//
+                .map(GeorchestraTargetConfig::headers)//
+                .ifPresent(mappings -> addJsonPayloads(exchange, mappings, headers));
+    }
 
-                        addJson(headers, "sec-user", mappings.getJsonUser(), user);
-                        addJson(headers, "sec-organization", mappings.getJsonOrganization(), org);
-                    });
-        };
+    private void addJsonPayloads(final ServerWebExchange exchange, final HeaderMappings mappings, HttpHeaders headers) {
+        Optional<GeorchestraUser> user = GeorchestraUsers.resolve(exchange);
+        Optional<Organization> org = GeorchestraOrganizations.resolve(exchange);
+        addJson(headers, "sec-user", mappings.getJsonUser().orElse(false), user);
+        addJson(headers, "sec-organization", mappings.getJsonOrganization().orElse(false), org);
     }
 
-    private void addJson(HttpHeaders target, String headerName, Optional<Boolean> enabled, Optional<?> toEncode) {
-        if (enabled.orElse(false)) {
+    private void addJson(HttpHeaders target, String headerName, boolean enabled, Optional<?> toEncode) {
+        if (enabled) {
```

Comment:
it looks more understandable to me to have the `orElse(false)` before calling the method, thanks for the readability improvement